src/controllers/tripRequestController.js

```diff
@@ -0,0 +1,173 @@
+
+            if (!trips) {
+                return res.status(404).json({ status: 'fail', message: `there are no Trip Requests assigned to ${userId}` })
+            } else {
+                return res.status(200).json({ status: 'success', data: trips })
+            }
+
+        } else if (req.user.role == 'manager') {
+            const trips = await tripRequests.findAll()
+
+            if (!trips) {
+                return res.status(404).json({ status: 'fail', message: `there are no Trip Requests found` })
+            } else {
+                return res.status(200).json({ status: 'success', data: trips })
+            }
+        } else {
+            return res.status(403).json({ status: 'fail', message: 'you are not allowed to retrieve a trip request' })
+        }
+    } catch (error) {
+        res.status(500).json({ status: 'error', message: 'error while getting trip request' })
+    }
+}
+
+export const updateTripRequest = async (req, res) => {
+    try {
+        if (req.user.role !== 'requester') {
+            return res.status(403).json({ status: 'fail', message: 'you are not allowed to update a trip request' })
+        }
+        const requestId = req.params.id;
+        const userId = req.user.id;
+
+        const tripRequest = await tripRequests.findOne({
+            where: { id: requestId }
+        })
+
+
+        if (!tripRequest) {
+            return res.status(404).json({ status: 'fail', message: `No Trip Request with id = ${requestId}` })
+        }
+
+        if (tripRequest.status == 'pending') {
+            await tripRequestSchema.validateAsync(req.body);
+            const type = req.body.returnDate == null ? 'one way trip' : 'return type trip';
+            const returnDate = type == 'one way trip' ? null : req.body.returnDate;
+
+            const updatedTrip = {
+                leavingFrom: req.body.leavingFrom,
+                goingTo: req.body.goingTo,
+                travelDate: req.body.travelDate,
+                returnDate: returnDate,
+                travelReason: req.body.travelReason,
+                tripType: type,
+                status: req.body.stastus,
+                requesterId: req.user.id,
+                accomodationId: req.body.accomodationId
+            }
+
+            tripRequests.update(updatedTrip, { where: { id: requestId, requesterId: userId } })
+                .then(num => {
+                    if (num == 1) {
+                        res.send({ status: 'success', message: 'trip request updated successfully.' })
+                    } else {
+                        res.send({
+                            message: `Cannot update trip request with id=${requestId}.`
+                        });
+                    }
+                })
+        } else {
+            return res.status(404).json({ status: 'fail', message: `Trip Request with id = ${requestId} is approved or rejected` })
```

Comment:
Or trip request does not exist, deleted,...
the message should be "Trip request with id: ${requestId} Not Found!" (my suggestion)

Reply:
on this part: if condition is checking whether tripRequest status is in pending status otherwise is rejected or approved. ( look previouse if condition for where trip request is not found/deleted)